backend/app/utils/request_parse.py

```diff
@@ -69,18 +70,24 @@ def get_location_offline(ip: str) -> list[str] | None:
 async def parse_ip_info(request: Request) -> tuple[str, str, str, str]:
     country, region, city = None, None, None
     ip = await get_request_ip(request)
-    if settings.LOCATION_PARSE == 'online':
-        location_info = await get_location_online(ip, request.headers.get('User-Agent'))
-        if location_info:
-            country = location_info.get('country')
-            region = location_info.get('regionName')
-            city = location_info.get('city')
-    elif settings.LOCATION_PARSE == 'offline':
-        location_info = await get_location_offline(ip)
-        if location_info:
-            country = location_info[0] if location_info[0] != '0' else None
-            region = location_info[2] if location_info[2] != '0' else None
-            city = location_info[3] if location_info[3] != '0' else None
+    location = await redis_client.get(f'{settings.IP_LOCATION_REDIS_PREFIX}:{ip}')
+    if location:
+        country, region, city = location.split(' ')
+    else:
+        if settings.LOCATION_PARSE == 'online':
```

Comment:
Do you want to try match-case?

Reply:
Match-case doesn't seem to be appropriate here?

Reply:
yes.